be/src/runtime/fragment_mgr.cpp

```diff
@@ -411,35 +420,24 @@ Status FragmentMgr::exec_plan_fragment(const TExecPlanFragmentParams& params, co
     RETURN_IF_ERROR(_exec_env->query_pool_mem_tracker()->check_mem_limit("Start execute plan fragment."));
 
     const TUniqueId& fragment_instance_id = params.params.fragment_instance_id;
-    std::shared_ptr<FragmentExecState> exec_state;
-    {
-        std::lock_guard<std::mutex> lock(_lock);
-        auto iter = _fragment_map.find(fragment_instance_id);
-        if (iter != _fragment_map.end()) {
-            // Duplicated
-            return Status::OK();
-        }
+    if (IsFragmentExecExist(fragment_instance_id)) {
+        return Status::OK();
     }
-    exec_state.reset(new FragmentExecState(params.params.query_id, fragment_instance_id, params.backend_num, _exec_env,
-                                           params.coord));
+    std::shared_ptr<FragmentExecState> exec_state = std::make_shared<FragmentExecState>(
+            params.params.query_id, fragment_instance_id, params.backend_num, _exec_env, params.coord);
     RETURN_IF_ERROR_WITH_WARN(exec_state->prepare(params), "Fail to prepare Fragment");
-
-    {
-        std::lock_guard<std::mutex> lock(_lock);
-        auto iter = _fragment_map.find(fragment_instance_id);
-        if (iter != _fragment_map.end()) {
-            // Duplicated
-            return Status::InternalError("Double execute");
-        }
-        // register exec_state before starting exec thread
-        _fragment_map.insert(std::make_pair(fragment_instance_id, exec_state));
+    // Check fragment exec whether exist again.
+    if (IsFragmentExecExist(fragment_instance_id)) {
+        return Status::InternalError("Double execute");
     }
+    // Register exec_state before starting exec thread.
+    _fragment_map.insert(std::make_pair(fragment_instance_id, exec_state));
```

Comment:
fragment_map insert without lock, may be dangerous

Reply:
OK, thanks.